src/agent/file_utils.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef DEVTOOLS_CDBG_DEBUGLETS_JAVA_FILE_UTILS_H_
+#define DEVTOOLS_CDBG_DEBUGLETS_JAVA_FILE_UTILS_H_
+
+#include "common.h"
+
+namespace devtools {
+namespace cdbg {
+
+/**
+ * Utility class to create a temporary directory for used by tests. Upon
+ * destruction it will delete the temporary directory and all of its contents
+ * (all files and all subdirectories). So the instance needs to be kept alive
+ * for the duration of the test.
+ */
+class TempPath {
+ public:
+  TempPath();
+  ~TempPath();
+
+  std::string path() const;
+
+ private:
+  string temp_path_;
+
+  DISALLOW_COPY_AND_ASSIGN(TempPath);
```

Comment:
Nit: This gives better error messages if it's in the public section or if you just explicitly delete the copy and assign operations.

Reply:
Done.